bounter/count_min_sketch.py

```diff
@@ -103,144 +96,85 @@ def __init__(self, size_mb=64, width=None, depth=None, algorithm='conservative')
+        return width * depth * CountMinSketch.cell_size(algorithm)
 
     def increment(self, key):
-        self.inc_method(key)
-        self.distinct_counter.add(str(key))
-        self.sum += 1
+        self.cms.increment(key)
+
+    def __getitem__(self, key):
+        return self.cms.get(key)
 
     def cardinality(self):
         """
         Returns an estimate for the number of distinct keys counted by the structure. The estimate should be within 1%.
         """
-        return int(self.distinct_counter.cardinality())
-
-    def basic_increment(self, key):
-        for row in range(self.depth):
-            self.table[row][self.bucket(row, key)] += 1
-
-    def conservative_increment(self, key):
-        current = self[key]
-        for row in range(self.depth):
-            bucket = self.bucket(row, key)
-            if self.table[row, bucket] == current:
-                self.table[row, bucket] += 1
-
-    def log_increment(self, key):
-        table = self.table
-        for row in range(self.depth):
-            bucket = self.bucket(row, key)
-            current = int(table[row, bucket])
-            if current < 2 * self.logbase:
-                current += 1
-                self.table[row, bucket] = current
-            else:
-                mask = (1 << (current // self.logbase - 1)) - 1
-                r = random.getrandbits(64)
-                if (r & mask) == 0:
-                    current += 1
-                    table[row, bucket] = current
-
-    def log_increment_conservative(self, key):
-        table = self.table
-        buckets = [self.bucket(row, key) for row in range(self.depth)]
-        values = [int(table[row, bucket]) for row, bucket in enumerate(buckets)]
-        current = min(values)
-        if current > 2 * self.logbase:
-            mask = (1 << (current // self.logbase - 1)) - 1
-            r = random.getrandbits(64)
-            if (r & mask) != 0:
-                return
-
-        for row, value in enumerate(values):
-            if value == current:
-                table[row, buckets[row]] = current + 1
-
-    def log_decode(self, log_value):
-        if (log_value <= 2 * self.logbase):
-            return log_value
-        else:
-            base = 1 << ((log_value >> (self.logshift + 1)) + self.logshift)
-            return int(base + (log_value % self.logbase) * (base / self.logbase))
+        return self.cms.cardinality()
 
-    def log_encode(self, input):
-        value = int(input)
-        if value <= 2 * self.logbase:
-            return value
-        base_bits = value.bit_length() - 1
-        log_value_base = (base_bits - self.logshift) * self.logbase
+    def total(self):
+        return self.cms.total()
 
-        remainder = (value >> (base_bits - self.logshift - 1)) - self.logbase
-        return log_value_base + remainder
+    def merge(self, other):
+        """
+        Merges another Count-min sketch structure into this one. The other structure must be initialized
```

Comment:
PEP8: In Python, docstrings are in imperative mode: `Do X, compute Y` (not `Does X, computes Y`). Here and elsewhere.